### PR TITLE
#228 feat: implementing sorting for library list in sidebar

### PR DESCRIPTION
**Pull request by Barraki**
[Task](https://github.com/orgs/hydralauncher/projects/1/views/1?pane=issue&itemId=62142229)

## Description

- Add `DropDownMenu` component
- Add logic for sorting library list in the sidebar
- Add localization for `en`, `ru`, `ua` lang for sorting list labels

Video:

https://github.com/hydralauncher/hydra/assets/33422772/d36986d1-f7de-46c1-9d1c-8160410a3b44

### CODE REVIEW

File: src/renderer/src/assets/sort-icon.svg
Comment (zamitto) [inline src/renderer/src/assets/sort-icon.svg:1]: no need to add this icon manually
there are suitable icons for sorting in the library that is already integrated in the project

https://primer.github.io/octicons/

File: src/renderer/src/hooks/use-event-listener.ts
Comment (zamitto) [inline src/renderer/src/hooks/use-event-listener.ts:1]: okay, this file sure looks like a copy paste from the internet. there seems to be no need for all these types and functions

Comment (Barraki) [reply src/renderer/src/hooks/use-event-listener.ts:1]: This hook from [usehooks-ts](https://usehooks-ts.com/react-hook/use-event-listener) library.

File: src/renderer/src/assets/sort-icon.svg
Comment (Barraki) [reply src/renderer/src/assets/sort-icon.svg:1]: Replace the icon from the library. https://primer.github.io/octicons/sort-desc-16

File: src/renderer/src/hooks/use-event-listener.ts
Comment (Barraki) [reply src/renderer/src/hooks/use-event-listener.ts:1]: - Refactored `use-on-click-outside` hooks and rewrote listener for detecting click-outside 
- Removed `use-event-listener` hook

**Pull request closed without merging**